frontend/src/components/DeleteTicketButton.js

```diff
@@ -0,0 +1,122 @@
+        }
+
+        if (currentTicket.type === TICKET_TYPES.FI) {
+            const FITree = currentTree.fundingItems.find(
+                (fi) => fi._id === currentTicket.id
+            )
+            return (
+                <Flex flexDir="column" m="16px 0">
+                    <Text>{`• ${currentTicket.codename}`}</Text>
+                    {FITree.personalPurchases.map((ppr) => (
+                        <Text
+                            key={ppr.code}
+                            pl="24px"
+                        >{`• ${ppr.codename}`}</Text>
+                    ))}
+                    {FITree.uwFinancePurchases.map((upr) => (
+                        <Text
+                            key={upr.code}
+                            pl="24px"
+                        >{`• ${upr.codename}`}</Text>
+                    ))}
+                </Flex>
+            )
+        }
+
+        // SF
+        return (
+            <Flex flexDir="column" m="16px 0">
+                <Text>{`• ${currentTicket.codename}`}</Text>
+                {currentTree.fundingItems.map((fi) => (
+                    <Box key={fi.code}>
+                        <Text pl="24px">{`• ${fi.codename}`}</Text>
+                        {fi.personalPurchases.map((ppr) => (
+                            <Text
+                                key={ppr.code}
+                                pl="48px"
+                            >{`• ${ppr.codename}`}</Text>
+                        ))}
+                        {fi.uwFinancePurchases.map((upr) => (
+                            <Text
+                                key={upr.code}
+                                pl="48px"
+                            >{`• ${upr.codename}`}</Text>
+                        ))}
+                    </Box>
+                ))}
+            </Flex>
+        )
+    }
+
+    return (
+        <AlertDialog
+            isOpen={isOpen}
+            leastDestructiveRef={cancelRef}
+            onClose={onClose}
+        >
+            <AlertDialogOverlay>
+                <AlertDialogContent>
+                    <AlertDialogHeader fontSize="lg" fontWeight="bold">
+                        Delete Current Ticket
+                    </AlertDialogHeader>
+
+                    <AlertDialogBody>
+                        <Text>The following tickets will be deleted:</Text>
+                        {getCascadeDeleteTicketCodenames()}
+                        <Text>
+                            Are you sure you want to delete these tickets?
+                        </Text>
+                    </AlertDialogBody>
+
+                    <AlertDialogFooter>
+                        <Button ref={cancelRef} onClick={onClose}>
+                            Cancel
+                        </Button>
+                        <Button
+                            colorScheme="red"
+                            onClick={() => {
+                                onClose()
+                                onDelete()
```

Comment:
you can add this now or I can add it later, but we should make this redirect to root as well as the ticket will be gone.

Reply:
onDelete() should handle that logic